mode/local_connection_extractor.go

```diff
@@ -90,6 +91,11 @@ func parseConnectionInfo(s *terraform.InstanceState) (*connectionInfo, error) {
 		connInfo.TimeoutVal = DefaultTimeout
 	}
 
+	if connInfo.PrivateKey != "" {
+
+		block, _ := pem.Decode([]byte(connInfo.PrivateKey))
+		connInfo.PrivateKey = string(pem.EncodeToMemory(block))
```

Comment:
Is it possible to check here what has been decoded? The example here: https://golang.org/pkg/encoding/pem/#example_Decode suggests so. Would that be a good idea?

Reply:
Good idea how does this look?